### ISSUE TITLE
Unknown modifier 'T'

### ISSUE THREAD

**Issue by queeniema** [ua-parser/uap-php#11]
Here is my PHP code based off of the sample usage code:
```
require_once($CFG->dirroot.'/vendor/autoload.php');
$ua = $_SERVER['HTTP_USER_AGENT'];
$parser = UAParser\Parser::create();
$result = $parser->parse($ua);
```

But I'm getting the following error message. Has anyone seen the same message before or know what the issue could be?

```
Warning: preg_match(): Unknown modifier 'T' in /vagrant/moodle/vendor/ua-parser/uap-php/src/AbstractParser.php on line 83
Call Stack
#	Time	Memory	Function	Location
1	0.0000	236840	{main}( )	../index.php:0
2	0.1604	11097280	create_help_message( )	../index.php:147
3	0.1635	12078336	UAParser\Parser->parse( )	../help_lib.php:349
4	0.1653	12097032	UAParser\DeviceParser->parseDevice( )	../Parser.php:51
5	0.1654	12099768	UAParser\AbstractParser->tryMatch( )	../DeviceParser.php:26
6	0.1656	12158752	preg_match ( )	../AbstractParser.php:83
```

**Comment by lstrojny** [comment]
Which PHP version are you using?

**Comment by queeniema** [comment]
I fixed it by downloading and copying the Github version of the regex php file into my resources directory. It seems like the update command didn't get the correct version of the file.

**Comment by lstrojny** [comment]
Thanks for bringing this further. Reopening the issue so I can have a look at the most recent version and what breaks there.

**Comment by JustBlackBird** [comment]
As I see the reason of the problem is in "@" character in the `regexex.yaml` because it is used by the library as the delimiter in PHP's regexes. Here is the problem line: https://github.com/ua-parser/uap-core/blob/master/regexes.yaml#L1235

The escaping of "@" character was added in 8a996f941984d895de0b9517707c2e3d7bfd127f but that changes was not published to `packagist.org`. That's why the version from repository works fine and the composer package does not.

**Comment by lstrojny** [comment]
Released 3.4.1 to address the issue. @JustBlackBird thanks for the reminder to ship an update.

**Comment by queeniema** [comment]
Thank you for the fix!
